features/facebookImport/src/components/activitiesMinistory/activitiesMinistory.jsx

```diff
@@ -0,0 +1,167 @@
+            {rangeIndex != 0 ? (
+                <button
+                    className="arrow left"
+                    onClick={() => onYearChange(yearRange[rangeIndex - 1])}
+                >
+                    <img src="./images/angle-left.svg" alt="arrow-left" />
+                </button>
+            ) : (
+                <div className="arrow filler"></div>
+            )}
+            <div className="year">{year}</div>
+            {rangeIndex != yearRange.length - 1 ? (
+                <button
+                    className="arrow right"
+                    onClick={() => onYearChange(yearRange[rangeIndex + 1])}
+                >
+                    <img src="./images/angle-right.svg" alt="arrow-right" />
+                </button>
+            ) : (
+                <div className="arrow filler"></div>
+            )}
+        </div>
+    );
+};
+
+const ActivitiesMinistory = ({ totalEvents }) => {
+    const yearRange = fillArray(Object.keys(totalEvents.values));
+
+    const yearlyTotals = Object.fromEntries(
+        yearRange.map((year) => [
+            year.toString().substring(2, 4),
+            totalEvents.values[year]?.total || 0,
+        ])
+    );
+
+    const [selectedYear, setSelectedYear] = useState(
+        yearRange[yearRange.length - 1]
+    );
+
+    const monthlyTotals = Object.fromEntries(
+        new Array(12)
+            .fill(undefined)
+            .map((_, index) => [
+                monthsAbbreviation[index],
+                totalEvents.values[selectedYear]?.values[index] || 0,
+            ])
+    );
+
+    const tabs = [
+        {
+            id: "total",
+            translation: i18n.t("activitiesMinistory:tab.total"),
+        },
+        {
+            id: "yearly",
+            translation: i18n.t("activitiesMinistory:tab.year"),
+        },
+    ];
+
+    const [activeTab, setActiveTab] = useState(tabs[0]);
+    const constantTabData = {
+        total: {
+            barData: yearlyTotals,
+            barWidth: "10px",
+            barChartLegendText: i18n.t("activitiesMinistory:tab.events.total", {
+                number_events: totalEvents.total,
+            }),
+        },
+        yearly: {
+            barData: monthlyTotals,
+            barWidth: "18px",
+            barChartLegendText: i18n.t(
+                "activitiesMinistory:tab.events.yearly",
+                {
+                    number_events: totalEvents.values[selectedYear]?.total || 0,
+                }
+            ),
+        },
+    }[activeTab.id];
+
+    //Adapt to oder of magnitude
+    const orderOfMagintude = (() => {
+        const highest = Math.max(...Object.values(constantTabData.barData));
+        if (highest >= 1000) {
+            constantTabData.barData = Object.fromEntries(
+                Object.entries(constantTabData.barData).map(([key, value]) => [
+                    key,
+                    value / 1000,
+                ])
+            );
+            return 1000;
```

Comment:
So bars are divided by 1000 if they exceed that value. The problem I see here is that in many cases, values will follow a power law, meaning that there will be some days exceeding 1000 and others close to 0. All those will be squashed to the bottom by this proportion. Point being, maybe consult with the UX experts and use logarithms here instead of simply dividing by 1000?
(Of course, another possibility is dividing by 10... Or systematically dividing by the highest and adjusting the y axis, which is the usual thing to do)

Reply:
It is clear they want to have thousands here to be able to just reduce by thousand, which makes sense from an UI point of view (text above says thousands and "ten-thousands" would look and feel quite bad). I asked them regarding logarithmic scales before and they weren't too thrilled as normal users don't get whats going on there, which makes sense. Also I do not think the difference between the months for the normal facebook user is going to be that great (you can see in my data at the top that it's quite homogenous)